spec/requests/api/orders/update_spec.rb

```diff
@@ -0,0 +1,44 @@
+RSpec.describe 'Update an order with a PUT /api/orders/:id', type: :request do
+
+  
+  let(:user) { create(:user) }
+  let(:product_1) { create(:product, name: 'Tshirt') }
+  let(:product_2) { create(:product, name: 'Football') }
+  let(:product_3) { create(:product, name: 'Scarf') }
+  let(:order) { create(:order) }
+  let!(:order_item_1) { create(:order_item, order:, product: product_2) }
+  let!(:order_item_2) { create(:order_item, order:, product: product_3) }
+
+  subject { response }
+  
+  describe 'succesful order' do
+  before do
+    put "/api/orders/#{order.id}", params: { product_id: product_2.id }
+
+    @order = Order.last
+  end
+
+  it { is_expected.to have_http_status :ok }
+
+  it 'response with an order confirmation' do
+    response_body = JSON.parse(response.body)
+    expect(response_body['message']).to eq 'Football added'
```

Comment:
`response_json` does the same thing with less code

Reply:
<img width="722" alt="Screenshot 2022-05-04 at 18 45 58" src="https://user-images.githubusercontent.com/92160782/166731748-b90e06cf-6aa8-4181-8b4a-a1391f70eb0e.png">